modules/calib3d/src/circlesgrid.cpp

```diff
@@ -839,7 +839,7 @@ Mat CirclesGridFinder::rectifyGrid(Size detectedGridSize, const std::vector<Poin
   //Mat H = findHomography( Mat( corners ), Mat( dstPoints ) );
 
   if (H.empty())
-      H = Mat::zeros(3, 3, CV_64FC1);
```

Comment:
I think in case of `findHomography` failure `warpedKeypoints` are useless at all, so amount of work can be reduced:
```
if (H.empty())
  {
      H = Mat::zeros(3, 3, CV_64FC1);
      warpedKeypoints.clear();
      return H;
  }
```